woocommerce/woocommerce
# Add defensive checks to `wc_get_order_status_name()`

As documented in the [linked issue](https://github.com/woocommerce/woocommerce/issues/51690), we've seen that fatal errors can occur if `wc_get_order_status_name()` is called with something other than a string (such as `null`).

This change tries to mitigate that, and also seeks to inform developers/site operators of any problems by issuing a *doing it wrong* notice if either of the following are true:

- The provided status is unknown/unregistered.
- The provided status is not a string.

Notes:

- PHP can coerce some types (like integers) into strings without any problem, but other types like arrays will lead to notices, warnings or errors being issued (the severity level depends on the PHP runtime being used). My stance is that is fine (we shouldn't seek to prevent those from taking place, we just want to avoid fatal errors).
- There are a few cases where a fatal error still happens, such as if an object is supplied instead of a string. I think it would also be possible to catch this and handle it gracefully ... but this is such an odd case that I don't think we *should.* We also don't know of any real-world cases where this is happening.

Closes #51690.



### How to test the changes in this Pull Request:

Real-world testing can be done in the following ways:

1. Install and activate Query Monitor, or ensure you have some other means of detecting errors and problems (including *doing it wrong* notices).
2. Ensure you have some order data.
3. Visit **WooCommerce ‣ Orders**. There should be no errors that map back to the `wc_get_order_status_name()` function (this function will be called multiple times when this screen is rendered, so this test is just to give us additional confidence that nothing has broken as a result of this change).

You can also do some console-based testing via `wp shell`. Examples:

```sh
>>> wc_get_order_status_name( 'pending' )
=> "Pending payment"
```

```sh
>>> wc_get_order_status_name( 'unknown' )
PHP Notice:  Function wc_get_order_status_name was called <strong>incorrectly</strong>. An invalid order status slug was supplied. 
=> "unknown"
```

```sh
>>> wc_get_order_status_name( 5 )
PHP Notice:  Function wc_get_order_status_name was called <strong>incorrectly</strong>. An invalid order status slug was supplied. 
=> "5"
```

```sh
>>> wc_get_order_status_name( null )
PHP Notice:  Function wc_get_order_status_name was called <strong>incorrectly</strong>. An invalid order status slug was supplied.
=> ""
```

```sh
>>> wc_get_order_status_name( [] )
PHP Warning:  Array to string conversion
PHP Notice:  Function wc_get_order_status_name was called <strong>incorrectly</strong>. An invalid order status slug was supplied. 
=> "Array"
```



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
